Add auto-update using GitHub Actions

johnhaddon opened the pull request:
This adds a GitHub workflow that periodically checks for new releases on GafferHQ/gaffer, automatically opening a pull request to update the documentation when necessary. An example is https://github.com/GafferHQ/documentation/pull/14, which was created in response to me pushing the branch for this PR.

themissingcow commented on update.py:
```diff
@@ -0,0 +1,115 @@
+
+# Update index.html to redirect to latest version
+
+with open( "index.html.in" ) as indexIn :
```
fwiw, I think Python allows multiple context managers per `with` clause.

johnhaddon replied:
Used in 97a9c84.

themissingcow commented on .github/workflows/update.yml:
```diff
@@ -0,0 +1,42 @@
+      run: |
+        ./update.py --set-outputs 1
+
+    - name: Create Pull Request
```
Is this relying on `create-pull-request` to be a no-op if there are no changes?

johnhaddon replied:
Correct, and that is the documented behaviour : 

> If there are no changes (i.e. no diff exists with the checked-out base branch), no pull request will be created and the action exits silently.

themissingcow commented on index.html:
```diff
@@ -2,10 +2,10 @@
-    <link rel="canonical" href="0.57.5.0/index.html">
-    <script>location="0.57.5.0/index.html"</script>
-    <meta http-equiv="refresh" content="0; url=0.57.5.0/index.html">
+    <link rel="canonical" href="0.59.0.0/index.html">
```
Should this change be part of this PR?

johnhaddon replied:
It should not - thanks for spotting - removed in 97a9c84.

themissingcow commented on update.py:
```diff
@@ -0,0 +1,115 @@
+
+# # If its up to date with the latest documented version, then we're done
+
+latestDocVersion = max( [ __versionFromString( v ) for v in glob.glob( "*.*.*.*" ) ] )
```
Given our past experience with failed deploys, Is it worth us checking how long `glob.glob(...)` is, and warning/erroring if there are more than _n_ versions already present.

johnhaddon replied:
Definitely, but I deliberately didn't tackle that this time. I will automate next time we hit the problem, when we can figure out what `n` should be (I presume it's not versions, but overall number of files or file size).

themissingcow reviewed (commented):
This is going to really help, nice one.

johnhaddon committed:
Add auto-update using GitHub Actions

johnhaddon committed:
Update from Gaffer 0.59.0.0

Using documentation from https://github.com/GafferHQ/gaffer/releases/download/0.59.0.0/gaffer-0.59.0.0-linux-python2.tar.gz

johnhaddon committed:
Merge pull request #14 from GafferHQ/create-pull-request/patch

Update from Gaffer 0.59.0.0
